app/modules/ws2812.c

```diff
@@ -387,24 +462,59 @@ static int ws2812_buffer_set(lua_State* L) {
 }
 
 static int ws2812_buffer_size(lua_State* L) {
-  ws2812_buffer * buffer = (ws2812_buffer*)lua_touserdata(L, 1);
-
-  luaL_argcheck(L, buffer && buffer->canary == CANARY_VALUE, 1, "ws2812.buffer expected");
+  ws2812_buffer * buffer = (ws2812_buffer*)luaL_checkudata(L, 1, "ws2812.buffer");
 
   lua_pushnumber(L, buffer->size);
 
   return 1;
 }
 
+static int ws2812_buffer_tostring(lua_State* L) {
+  ws2812_buffer * buffer = (ws2812_buffer*)luaL_checkudata(L, 1, "ws2812.buffer");
+
+  luaL_Buffer result;
```

Comment:
Does this implementation work for RGB chains longer than 85 (aka `LUAL_BUFFERSIZE / 3`)?